samples/joke/README.md

```diff
@@ -14,44 +14,179 @@ cluster. You can just follow the steps below to get started quickly:
+
+```shell
+opm alpha bundle generate \
+    --directory target/manifests \
+    --package <The name of the package that bundle image belongs to> \ 
+    --channels <The list of channels that bundle image belongs to> \
+    --default <The default channel for the bundle image>
+```
+ 
+For example, if we want to name the package `joke` and use the `alpha` channels, the command would look like as `opm alpha bundle generate --directory target/manifests --package joke --channels alpha --default alpha`.
+
+---
+**NOTE**
+Find more information about channels and packages in [this link](https://olm.operatorframework.io/docs/best-practices/channel-naming/#channels).
+---
+
+The above command will generate a `bundle.Dockerfile` Dockerfile file that you will use to build and push the final Operator Bundle image to your container registry:
+
+```shell
+JOKE_BUNDLE_IMAGE=<your container registry>/<your container registry namespace>/joke-manifest-bundle:<tag>
+docker build -t $JOKE_BUNDLE_IMAGE -f bundle.Dockerfile .
+docker push $JOKE_BUNDLE_IMAGE
+```
+
+For example, if our container registry is `quay.io`, our Quay user is `myuser` and the tag we're releasing is `1.0`, the final `JOKE_BUNDLE_IMAGE` property would be `quay.io/myuser/joke-manifest-bundle:1.0`. 
+
+3. Make your operator available within a Catalog
+
+OLM uses catalogs to discover and install Operators and their dependencies. So, a catalog is similar to a repository of operators and their associated versions that can be installed on a cluster. Moreover, the catalog is also a container image that contains a collection of bundles and channels. Therefore, we'd need to create a new catalog (or update an existing one if you're already have one), build/push the catalog image and then install it on our cluster.
+
+So far, we have already built the Operator bundle image at `$JOKE_BUNDLE_IMAGE` (see above) and next, we need to add this Operator bundle image into our catalog. For doing this, we'll use the `olm` tool as follows:
+
+```shell
+CATALOG_IMAGE=<catalog container registry>/<catalog container registry namespace>/<catalog name>:<tag>
+opm index add \
+    --bundles $JOKE_BUNDLE_IMAGE \
+    --tag $CATALOG_IMAGE \
+    --build-tool docker
+docker push $CATALOG_IMAGE
+```
+
+For example, if our catalog name is `joke-catalog`, our container registry for the catalog is `quay.io`, our Quay user is `myuser` and the container tag we're releasing is `59.0`, the final `CATALOG_IMAGE` property would be `quay.io/myuser/joke-catalog:59.0`.
+
+---
+**NOTE**
+If you're using an insecure registry, you'd need to append the argument `--skip-tls` to the `opm index` command.
+---
+
+Once we have our catalog image built and pushed at `$CATALOG_IMAGE` image, we need to install it on our cluster using the `CatalogSource` resource by doing the next command:
```

Comment:
It's quite weird to need to create an image to create a catalog source but I guess there are reasons for this… 🤷🏼

Reply:
Maybe we can enhance this process in a later pull request.